src/basic/compress.c

```diff
@@ -65,6 +65,21 @@ static const char* const compression_table[_COMPRESSION_MAX] = {
 
 DEFINE_STRING_TABLE_LOOKUP(compression, Compression);
 
+bool compression_supported(Compression c) {
+        switch (c) {
+        case COMPRESSION_NONE:
+                return true;
+        case COMPRESSION_XZ:
+                return HAVE_XZ;
+        case COMPRESSION_LZ4:
+                return HAVE_LZ4;
+        case COMPRESSION_ZSTD:
+                return HAVE_ZSTD;
+        default:
+                return false;
+        }
+}
```

Comment:
could also do:

```c
static const unsigned supported = 
        (1U << COMPRESSION_NONE) |
        (1U << COMPRESSION_XZ) * HAVE_XZ |
        (1U << COMPRESSION_LZ4) * HAVE_LZ4 |
        (1U << COMPRESSION_ZSDT) * HAVE_ZSTD;

return c >= 0 && c < _COMPRESSION_MAX && FLAGS_SET(supported, 1U << c);
```

but dunno, up to you.

Reply:
That'd be just one instruction, instead of a few for the switch stmt. Seems like a nice optimization ;)